src/modules/edc-demo/components/contract-agreement-transfer-dialog/contract-agreement-transfer-dialog.component.ts

```diff
@@ -65,23 +76,47 @@ export class ContractAgreementTransferDialog implements OnDestroy {
   private buildDataAddressDto(
     formValue: ContractAgreementTransferDialogFormValue,
   ): DataAddressDto {
-    switch (formValue.dataAddressType) {
+    switch (formValue.datasink?.dataAddressType) {
       case 'Custom-Data-Address-Json':
-        return JSON.parse(formValue.dataDestination?.trim()!!);
+        return JSON.parse(formValue.datasink?.dataDestination?.trim()!!);
       case 'Http':
+        const httpDatasinkProperties =
+          this.httpDatasinkFormMapper.buildHttpDatasinkProperties(
+            formValue.datasink,
+          );
         return {
           properties: {
             type: 'HttpData',
-            baseUrl: formValue.httpUrl!!.trim(),
+            ...this.encodeHttpDatasinkProperties(httpDatasinkProperties),
           },
         };
       default:
         throw new Error(
-          `Invalid Data Address Type ${formValue.dataAddressType}`,
+          `Invalid Data Address Type ${formValue.datasink?.dataAddressType}`,
         );
     }
   }
 
+  private encodeHttpDatasinkProperties(
```

Comment:
We should have a common method "encodeHttpRequestParams". HttpDatasourceProperties and HttpDatasinkProperties are both HttpRequestParams in the backend, so there's no reason to keep them apart here.